### ISSUE TITLE
doubt about using labels of lane

### ISSUE THREAD

**Issue by SarchWu** [ucbdrive/bdd-data#33]
Hi，fisher：
 I was trying to use the labels and images of 100k for roadmarkings sgementation,such as lane of white or
yellow.But the annotaions of lanes objects are a series of lines which consist of points.I couldn't find any correlation between different lines based on label json.So i can't generate the lanes mask. Would you mind tell me that this dataset whether or not generate lane masks.If could,could you tell me how to generate it ?Thanks~

**Comment by hqhe** [comment]
hi,I am  trying to do the lane mask also,do you have any idea?

**Comment by abhigoku10** [comment]
@SarchWu @hqhe hey guys were you able to generate the masks for the lane markings ?? if so can you please share the work

**Comment by SarchWu** [comment]
@abhigoku10 @hqhe 
     Hey ,guys , I have read the official code released by fisher ,so I tried a method to generate the lane mask ,but still exist some flaw , hope this method could benifit you.
     The lane's annotations consists of two types ,the one with two points ,the other with three control point and one point . **It means that we could use Bezier curve or linear interpolation to fit the anno line .** 
    The lane's annotations are formed with two styles.The first is that each lane marked only with one line in the middle of lane ,the second is that each lane marked with two line along the boundry. **So after fitting the line ,the second thing we should do is to get the relationship between lines in the same image ,whether they belong to one lane or seperate to each other.** In this step, we could cal the distance between two lines ,or intersect area , or curvature ,this kind of information could help us. 
    After getting the relationship ,we could get a close polygon to generate the mask.

**Comment by abhigoku10** [comment]
@SarchWu  when i checked with the annotation lane marking i see the markings are way of so to obtain a smoother curve what do u suggest below is an example 
![image](https://user-images.githubusercontent.com/22094053/70514295-617f6680-1b59-11ea-9e8d-2d184ca1a1dd.png)

**Comment by SarchWu** [comment]
> @SarchWu when i checked with the annotation lane marking i see the markings are way of so to obtain a smoother curve what do u suggest below is an example
> ![image](https://user-images.githubusercontent.com/22094053/70514295-617f6680-1b59-11ea-9e8d-2d184ca1a1dd.png)

yes,the curve is perfect

**Comment by abhigoku10** [comment]
@SarchWu  so r u suggesting to use beizer curves or active contour methods for drawing , in the above image the yellow points are the annotation points and the blue is the lane , i have not still drawn anythign on the image

**Comment by SarchWu** [comment]
> @SarchWu so r u suggesting to use beizer curves or active contour methods for drawing , in the above image the yellow points are the annotation points and the blue is the lane , i have not still drawn anythign on the image
yes,for line consists of two points,use linear interplote to fit the line ,for line consists of four points ,user beizer curve to fit .
Then such as the lane in the example image will exist.
